aif360/algorithms/preprocessing/lfr.py

```diff
@@ -81,30 +83,35 @@ def fit(self, dataset, maxiter=5000, maxfun=5000, **kwargs):
-        nonsensitive_idx = np.array(np.where(d == self.privileged_group_protected_attribute_value))[0].flatten()
-        training_sensitive = dataset.features[sensitive_idx]
-        training_nonsensitive = dataset.features[nonsensitive_idx]
-        ytrain_sensitive = dataset.labels[sensitive_idx]
-        ytrain_nonsensitive = dataset.labels[nonsensitive_idx]
-
-        model_inits = np.random.uniform(size=self.features_dim * 2 + self.k + self.features_dim * self.k)
+        unprivileged_sample_ids = np.array(np.where(protected_attributes == self.unprivileged_group_protected_attribute_value))[0].flatten()
+        privileged_sample_ids = np.array(np.where(protected_attributes == self.privileged_group_protected_attribute_value))[0].flatten()
+        features_unpriveleged = dataset.features[unprivileged_sample_ids]
+        features_priveleged = dataset.features[privileged_sample_ids]
+        labels_unpriveleged = dataset.labels[unprivileged_sample_ids]
+        labels_priveleged = dataset.labels[privileged_sample_ids]
+
+        # Initialize the LFR optim objective parameters
+        parameters_initialization = np.random.uniform(size=self.k + self.features_dim * self.k)
         bnd = []
-        for i, _ in enumerate(model_inits):
+        for i, _ in enumerate(parameters_initialization):
             if i < self.features_dim * 2 or i >= self.features_dim * 2 + self.k:
                 bnd.append((None, None))
             else:
                 bnd.append((0, 1))
+        lfr_helpers.LFR_optim_objective.steps = 0
+
+        self.learned_model = optim.fmin_l_bfgs_b(lfr_helpers.LFR_optim_objective, x0=parameters_initialization, epsilon=1e-5,
+                                                      args=(features_unpriveleged, features_priveleged,
+                                        labels_unpriveleged[:, 0], labels_priveleged[:, 0], self.k, self.Ax,
+                                        self.Ay, self.Az, self.print_interval),
+                                                      bounds=bnd, approx_grad=True, maxfun=maxfun,
+                                                      maxiter=maxiter, disp=self.verbose)[0]
+        self.w = self.learned_model[:self.k]
+        self.prototypes = self.learned_model[self.k:].reshape((self.k, self.features_dim))
 
-        self.learned_model = optim.fmin_l_bfgs_b(lfr_helpers.LFR_optim_obj, x0=model_inits, epsilon=1e-5,
-                                  args=(training_sensitive, training_nonsensitive,
-                                        ytrain_sensitive[:, 0], ytrain_nonsensitive[:, 0], self.k, self.Ax,
-                                        self.Ay, self.Az, 0, self.print_interval),
-                                  bounds=bnd, approx_grad=True, maxfun=maxfun,
-                                  maxiter=maxiter, disp=self.verbose)[0]
         return self
 
     def transform(self, dataset, threshold=0.5, **kwargs):
```

Comment:
same thing here: can we remove `**kwargs`?

Reply:
done